unit/euclid.go

```diff
@@ -41,26 +41,28 @@ func (e *euclid) ProcessSample(i int) {
 		out    = -1.0
 	)
 
-	if e.lastSpan != span || e.lastFill != fill {
-		for i := range e.pattern {
-			e.counts[i] = 0
-			e.remainders[i] = 0
-			e.pattern[i] = false
+	if fill > 0 {
```

Comment:
Let's update this to return early for the exceptional case:

```go
if fill <= 0 {
    e.out.Write(i, out)
    return
}

// normal behavior goes here...
```

Reply:
Will also need to capture the `last*` -1 samples (e.g. `lastTrigger`) for this exceptional case in some way. I left that out of the snippet above.

Reply:
I like the idea of the early return, but isn't this a better approach?
```go
if fill > 0 {
	// normal behavior
}

e.lastSpan = span
e.lastFill = fill
e.lastIdx = e.idx
e.lastTrigger = trig
```

Reply:
Yeah that works fine.